dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/export/trackedentity/TrackedEntityOperationParamsMapper.java

```diff
@@ -140,16 +145,14 @@ public TrackedEntityQueryParams map(TrackedEntityOperationParams operationParams
     return params;
   }
 
-  private List<TrackedEntityType> getTrackedEntityTypes(
-      TrackedEntityType trackedEntityType, Program program, User user) throws BadRequestException {
+  private List<TrackedEntityType> getTrackedEntityTypes(Program program, User user)
```

Comment:
Based on the validation that we do to the parameters in the request, we should have 3 cases here:
- TET from the params
- TET from the Program (it should be validated already to be a tracker program, hence the TET cannot be null)
- TETs from the list of TrackedEntitites requested

We shouldn't have a case where we look at all the TET that the user can access

Reply:
Agreed with this one too, the problem here is that getting the list of accessible TET is the only way we have to validate the access to a tracked entity when no program nor TET are present in the request.
Refactoring this now could lead to changes in multiple places and I don't feel comfortable doing it so close to the release.
I'd rather do it in another ticket that just focuses on this part, and preferably, after the release.